ui/lib/core/addon/components/kv-suggestion-input.ts

```
@@ -0,0 +1,149 @@
+ * 
+ * @example
+ * <KvSuggestionInput
+    @label="Select a secret to sync"
+    @subText="Enter the full path to the secret. Suggestions will display below if permitted by policy."
+    @value={{this.secretPath}}
+    @mountPath={{this.mountPath}} // input disabled when mount path is not provided
+    @onChange={{fn (mut this.secretPath)}}
+  /> 
+ */
+
+interface Args {
+  label: string;
+  subText?: string;
+  mountPath: string;
+  value: string;
+  onChange: CallableFunction;
+}
+
+interface PowerSelectAPI {
+  actions: {
+    open(): void;
+    close(): void;
+  };
+}
+
+export default class KvSuggestionInputComponent extends Component<Args> {
+  @service declare readonly store: StoreService;
+
+  @tracked secrets: KvSecretMetadataModel[] = [];
+  powerSelectAPI: PowerSelectAPI | undefined;
+  _cachedSecrets: KvSecretMetadataModel[] = []; // cache the response for filtering purposes
+
+  constructor(owner: unknown, args: Args) {
+    super(owner, args);
+    if (this.args.mountPath) {
+      this.updateSuggestions();
+    }
+  }
+
+  get inputId() {
+    // add unique segment to id in case multiple instances of component are used on the same page
+    return `suggestion-input-${guidFor(this)}`;
+  }
+
+  async fetchSecrets(isDirectory: boolean) {
+    const { mountPath } = this.args;
+    try {
+      const backend = keyIsFolder(mountPath) ? mountPath.slice(0, -1) : mountPath;
+      const parentDirectory = parentKeyForKey(this.args.value);
+      const pathToSecret = isDirectory ? this.args.value : parentDirectory;
+      const kvModels = (await this.store.query('kv/metadata', {
+        backend,
+        pathToSecret,
+      })) as unknown;
+      // this will be used to filter the existing result set when the search term changes within the same path
+      this._cachedSecrets = kvModels as KvSecretMetadataModel[];
+      return this._cachedSecrets;
+    } catch (error) {
+      console.log(error); // eslint-disable-line
```

Do we want to keep this console.log? It seems like we should just silently error 🤔

---

I was running into troubles trying to debug an issue which was actually erroring here so I think it is helpful having it logged to the console, even for a user if they contacting support and are trying to gather information.